boxsdk/object/metadata_template.py

```diff
@@ -0,0 +1,282 @@
+        :type field:
+            :class:`MetadataField`
+        """
+        self.ops.append({
+            'op': 'editField',
+            'fieldKey': field_key,
+            'data': field.json(),
+        })
+
+    def edit_enum_option_key(self, field_key, old_option_key, new_option_key):
+        """
+        Change the key of an enum field option.
+
+        :param field_key:
+            The key of the template field in which the option appears
+        :type field_key:
+            `unicode`
+        :param old_option_key:
+            The old option key
+        :type old_option_key:
+            `unicode`
+        :param new_option_key:
+            The new option key
+        :type new_option_key:
+            `unicode`
+        """
+        self.ops.append({
+            'op': 'editEnumOption',
+            'fieldKey': field_key,
+            'enumOptionKey': old_option_key,
+            'data': {
+                'key': new_option_key,
+            },
+        })
+
+    def remove_enum_option(self, field_key, option_key):
+        """
+        Remove an option from an enum field.
+
+        :param field_key:
+            The key of the template field in which the option appears
+        :type field_key:
+            `unicode`
+        :param option_key:
+            The key of the enum option to remove
+        :type option_key:
+            `unicode`
+        """
+        self.ops.append({
+            'op': 'removeEnumOption',
+            'fieldKey': field_key,
+            'enumOptionKey': option_key,
+        })
+
+    def remove_field(self, field_key):
+        """
+        Remove a field from the metadata template.
+
+        :param field_key:
+            The key of the field to remove
+        :type field_key:
+            `unicode`
+        """
+        self.ops.append({
+            'op': 'removeField',
+            'fieldKey': field_key,
+        })
+
+
+class MetadataFieldType(TextEnum):
+    STRING = 'string'
+    DATE = 'date'
+    ENUM = 'enum'
+    MULTISELECT = 'multiSelect'
+    FLOAT = 'float'
+
+
+class MetadataField(object):
+    """Represents a metadata field when creating or updating a metadata template."""
+
+    def __init__(self, field_type, display_name, key=None, options=None):
+        self.type = field_type
+        self.name = display_name
+        self.key = key
+        self.options = options
+
+    def json(self):
+        """
+        Returns the correct representation of the temnplate field for the API.
+
+        :rtype:
+            `dict`
+        """
+        values = {}
+
+        if self.type is not None:
+            values['type'] = self.type
+
+        if self.name is not None:
+            values['displayName'] = self.name
+
+        if self.key is not None:
+            values['key'] = self.key
+
+        if self.type in ['enum', 'multiSelect']:
+            values['options'] = [{'key': opt} for opt in self.options] if self.options is not None else []
```

Comment:
could simplify to `values['options'] = [{'key': opt} for opt in self.options or ()]`